x/gamm/spec/01_concepts.md

```diff
@@ -4,5 +4,50 @@ order: 1
 
 # Concepts
 
-The `gamm` module currently provides interfaces for balancer v1-AMMs.
-This will evolve over time into a repository with a generalized AMM interface.
+The concept of the `gamm` module is designed to handle assets of a chain using the AMM and its concept of pool shares. 
+
+## Pool
+
+### Creation of Pool
+
+At an initial creation of the pool, a fixed amount of 100 share token is minted in the pool and sent to the creator of the pool's account. Pool share denom is in the format of gamm/pool/{poolId} and is displayed in the format of GAMM-{poolId} to the user. Pool assets are sorted in alphabetical order by defualt.
+
+### Joining Pool
+
+When joining a pool, users provide maximum amount of tokens willing to deposit, while the FE takes care of the calculation of how many share tokens the user is eligible at the specific moment of sending the transaction. Calculation of exactly how many tokens are needed to get the designated share is done at the moment of procssing the transaction, validating that it does not exceed the maximum amount of token the user is willing to deposit. After the validation, the share of the pool is minted and sent to the user account. Joining the pool using a single asset is also possible.
```

Comment:
FE?

Reply:
Oh I meant Front End in the context that the JoinPool function takes in how much share token as a parameter. From what I know, the calculation is done in the front-end, providing the function with the amount of share token the user is eligible for, while the user only provides the max-token. Do I have a misunderstanding somewhere?

Reply:
I think @ValarDragon just didn't realize what FE meant.  Probably just write out "frontend"

Reply:
Ah I see, fixed!